crates/wash-lib/src/build.rs

```diff
@@ -96,19 +99,48 @@ pub fn build_actor(
     common_config: &CommonConfig,
     signing_config: Option<SignConfig>,
 ) -> Result<PathBuf> {
-    // Build actor based on language toolchain
-    let actor_wasm_path = match language_config {
-        LanguageConfig::Rust(rust_config) => {
-            build_rust_actor(common_config, rust_config, actor_config)
-        }
-        LanguageConfig::TinyGo(tinygo_config) => {
-            build_tinygo_actor(common_config, tinygo_config, actor_config)
+    let actor_wasm_path = if let Some(raw_command) = common_config.build_command.as_ref() {
+        let mut split_command = raw_command.split_ascii_whitespace();
+        let command = split_command
+            .next()
+            .context("build command is supplied but empty")?;
+
+        // Change directory into the project directory
+        std::env::set_current_dir(&common_config.path)?;
+
+        let mut command = process::Command::new(command);
+        command.args(split_command).status().map_err(|e| {
```

Comment:
Readability nit: instead of continuing to use the mut iterator where we've consumed the first token, I would collect the rest of the iterator to an `args` var above, and pass that collection here. At first thought this would include the first token

Reply:
But but efficiency!

Settle for a middleground of a clarifying comment?

Reply:
I feel firmly that the readability is harmed, and the cost to allocate a vec of a few bytes in a CLI command is negligible, so a comment would make me think "change the source then?"

But that's just like, my opinion, man. And I respect your opinion, so if you disagree, I won't be upset 🤝